apps/aecore/test/aec_mining_tests.erl

```diff
@@ -31,10 +31,12 @@ mine_block_test_() ->
          fun() ->
                  TopBlock = #block{height = 0,
                                    target = ?HIGHEST_TARGET_SCI},
-                 meck:expect(aec_pow, pick_nonce, 0, 9),
-
-                 {ok, BlockCandidate, Nonce} = ?TEST_MODULE:create_block_candidate(TopBlock, []),
-                 {ok, Block} = ?TEST_MODULE:mine(BlockCandidate, Nonce),
+                 %% Try enough nonces to find a solution.
```

Comment:
It should be deterministic, or we don't want it as a test... We don't want to loop and try many nonces here.
We are using aec_test_utils:mock_fast_and_deterministic_cuckoo_pow() in the cuckoo case.